openmls/src/group/tests/group_context_extensions.rs

```diff
@@ -0,0 +1,1141 @@
+    let frankenstein::FrankenMlsMessage {
+        version,
+        body:
+            frankenstein::FrankenMlsMessageBody::PublicMessage(frankenstein::FrankenPublicMessage {
+                content:
+                    frankenstein::FrankenFramedContent {
+                        group_id,
+                        epoch: gce_commit_epoch,
+                        sender,
+                        authenticated_data,
+                        ..
+                    },
+                ..
+            }),
+    } = frankenstein::FrankenMlsMessage::from(gce_req_cap_commit)
+    else {
+        unreachable!()
+    };
+
+    let charlie = PartyState::<Provider>::generate("charlie", ciphersuite);
+    let charlie_kpb = charlie.key_package(ciphersuite, |builder| {
+        builder.leaf_node_capabilities(
+            Capabilities::builder()
+                .extensions(vec![ExtensionType::Unknown(0xf001)])
+                .build(),
+        )
+    });
+
+    let commit_content = frankenstein::FrankenFramedContent {
+        body: frankenstein::FrankenFramedContentBody::Commit(frankenstein::FrankenCommit {
+            proposals: vec![frankenstein::FrankenProposalOrRef::Proposal(
+                frankenstein::FrankenProposal::Add(frankenstein::FrankenAddProposal {
+                    key_package: charlie_kpb.key_package.into(),
+                }),
+            )],
+            path: None,
+        }),
+        group_id,
+        epoch: gce_commit_epoch + 1,
+        sender,
+        authenticated_data,
+    };
+
+    let group_context = alice.group.export_group_context().clone();
+
+    let bob_group_context = bob.group.export_group_context();
+    assert_eq!(
+        bob_group_context.confirmed_transcript_hash(),
+        group_context.confirmed_transcript_hash()
+    );
+
+    let secrets = alice.group.group().message_secrets();
+    let membership_key = secrets.membership_key().as_slice();
+
+    let franken_commit = frankenstein::FrankenMlsMessage {
+        version,
+        body: frankenstein::FrankenMlsMessageBody::PublicMessage(
+            frankenstein::FrankenPublicMessage::auth(
+                &alice.party.provider,
+                ciphersuite,
+                &alice.party.signer,
+                commit_content,
+                Some(&group_context.into()),
+                Some(membership_key),
+                // this is a dummy confirmation_tag:
+                Some(vec![0u8; 32].into()),
+            ),
+        ),
+    };
+
+    let fake_commit = MlsMessageIn::tls_deserialize(
+        &mut franken_commit.tls_serialize_detached().unwrap().as_slice(),
+    )
+    .unwrap();
+
+    // Note: If this starts failing, the order in which validation is checked may have changed and we
+    // fail on the fact that the confirmation tag is wrong. in that case, either the check has to be
+    // disabled, or the frankenstein framework needs code to properly commpute it.
+    let err = bob.fail_processing(fake_commit);
```

Comment:
```suggestion
    // disabled, or the frankenstein framework needs code to properly compute it.
```

Reply:
I don't think I understand this suggestion.

Reply:
Ah... vscode has a one-off bug where it uses the wrong line. There's a typo inline 527 ("commpute")

Reply:
Ahh, got it, thanks!